client/src/components/SignInSignUpModal.tsx

```diff
@@ -0,0 +1,59 @@
+// import { FormEvent, useEffect, useState } from 'react';
+import { AiOutlineClose } from 'react-icons/ai';
+
+type props = {
+  onClose: () => void;
+  signInStatus: boolean;
+};
+
+export default function SignInSignUpModal({ onClose, signInStatus }: props) {
```

Comment:
There will be two buttons on the navbar (sign in and sign up) the user can still see entries made by other people and navigate to people's profiles but won't be able to create a post or view their profile(because it doesnt exist) until they sign in. The modal shows up if the user presses either sign in or sign up and it will display above whatever current page they are looking at. If this is a bad approach let me know.